sled-agent/src/metrics.rs

```diff
@@ -100,6 +105,39 @@ impl LinkKind {
     const OPTE: &'static str = "opte";
 }
 
+struct Target {
+    id: TargetId,
+    sled_datalink: SledDataLink,
+}
+
+impl Target {
+    /// Create a new target.
+    fn new(id: TargetId, sled_datalink: SledDataLink) -> Self {
```

Comment:
Similar comment to earlier, it seems simpler to direct initialize these fields, since we always have access to them in this module.

Reply:
I had these initially during prototyping, but they're unnecessary.